test/appmon/appmon_test.go

```diff
@@ -0,0 +1,174 @@
+//go:build e2e
+
+package appmon
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+
+	"github.com/Dynatrace/dynatrace-operator/src/api/v1beta1"
+	"github.com/Dynatrace/dynatrace-operator/src/config"
+	"github.com/Dynatrace/dynatrace-operator/src/kubeobjects"
+	"github.com/Dynatrace/dynatrace-operator/src/kubeobjects/address"
+	"github.com/Dynatrace/dynatrace-operator/test/dynakube"
+	"github.com/Dynatrace/dynatrace-operator/test/kubeobjects/deployment"
+	"github.com/Dynatrace/dynatrace-operator/test/kubeobjects/environment"
+	"github.com/Dynatrace/dynatrace-operator/test/kubeobjects/manifests"
+	"github.com/Dynatrace/dynatrace-operator/test/kubeobjects/namespace"
+	"github.com/Dynatrace/dynatrace-operator/test/kubeobjects/pod"
+	"github.com/Dynatrace/dynatrace-operator/test/operator"
+	"github.com/Dynatrace/dynatrace-operator/test/sampleapps"
+	"github.com/Dynatrace/dynatrace-operator/test/secrets"
+	"github.com/Dynatrace/dynatrace-operator/test/webhook"
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/e2e-framework/pkg/env"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+)
+
+const (
+	installSecretPath = "../testdata/secrets/application-monitoring-install.yaml"
```

Comment:
why do we need a special file for every test, when its always the same ?

Reply:
because the secrets are not checked in and it doesn't have to be the case that they are always the same. Someone might have different tokens or tenants for different tests which requires different files.

however, that example is pretty contrived so I could also use only one file.

Reply:
we should decide on how we are doing this, because on the https://github.com/Dynatrace/dynatrace-operator/pull/1250 we have the single file way. 

we could have special files when its needed, and we could have samples for each type of file we need.
by doing this we would not bloat the repo with unnecessary amounts of files,

Reply:
merged the files